server/src/main/java/org/elasticsearch/index/query/SpanOrQueryBuilder.java

```diff
@@ -74,10 +75,23 @@ public SpanOrQueryBuilder addClause(SpanQueryBuilder clause) {
         if (clause == null) {
             throw new IllegalArgumentException("[" + NAME + "] inner clause cannot be null");
         }
+        checkBoostValue(clause);
```

Comment:
Similar comment here: I think this is not necessary. The only place where we should do the check is inside fromXContent method for all span query builders : SpanOrQueryBuilder, SpanTermQueryBuilder etc.